### ISSUE TITLE
Docstrings not reported as foldable elements by basedpyright language server

### ISSUE THREAD

**Issue by lolo120916** [DetachHead/basedpyright#761]
Hello,

Just a small bug report concerning the language server.

The basedpyright language server does not report docstrings as foldable elements, unlike the pylance language server. This inconsistent behavior can be confusing for users who expect docstrings to be foldable, especially for long ones.

When using the basedpyright language server, docstrings in Python code are not recognized as foldable elements, meaning that users cannot collapse or expand them in their code editor. This is in contrast to the pylance language server, which correctly identifies docstrings as foldable elements.

Reproduction Steps:

1. Create a Python file with a function that has a docstring, for example:

```python
def my_function(arg1, arg2):
    """
    This is a docstring for the my_function function.
    It provides a brief description of the function and its parameters.
    """
    # Function implementation
    pass
```

2. Open the file in a code editor (Vscode or Vscodium) that supports the basedpyright language server.
3. Observe that the docstring is not recognized as a foldable element, and cannot be collapsed or expanded.
4. Repeat the same steps, but this time use the pylance language server.
5. Observe that the docstring is correctly identified as a foldable element and can be collapsed or expanded.

Many thanks for the work you've done on the basedpyright language server, and I hope this bug report can help improve the overall user experience.

**Comment by DetachHead** [comment]
the language server currently doesn't do folding. the folding you see comes from vscode guessing based on the indentation

merging this with #308

**Comment by lolo120916** [comment]
Thank you for reclassifying my bug report and linking it to another issue. I look forward to any updates
